test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php

```diff
@@ -501,6 +501,83 @@ public function testStubForConstantThatIsDeprecated(): void
         self::assertSame('filter', $stubData->getExtensionName());
     }
 
+    public function testStubForConstantThatIsDeprecatedInPatchRelease(): void
+    {
+        // use a faked stub to make this test independent of the actual PHP version
+        $exampleStub = <<<'EOT'
+<?php
+
+/**
+ * @deprecated 8.1.2
+ */
+\define('A_CUSTOM_CONSTANT', 513);
+EOT;
+        $stubData    = new StubData($exampleStub, null);
+
+        self::assertStringMatchesFormat(
+            "%Adefine('A_CUSTOM_CONSTANT',%w%d);",
+            $stubData->getStub(),
+        );
+
+        if (PHP_VERSION_ID >= 80100) {
```

Comment:
ahh I think we don't catch a bug because we never reach the ELSE case of this condition (because CI is running only 8.2/8.3)

Reply:
Ah, good catch! Indeed! 

Perhaps we can change the fixture to introduce deprecations with higher/fictional PHP versions? :D

Reply:
Let me first make sure phpstan can catch such bugs in the future. Brb

Reply:
https://github.com/phpstan/phpstan-src/pull/2968 should help us in the future.